site/docs/types/type_classes.md

```diff
@@ -31,16 +31,18 @@ Simple type classes are those that don't support any form of configuration. For
+| DECIMAL&lt;P, S&gt;           | A fixed-precision decimal value having precision (P, number of digits) <= 38 and scale (S, number of fractional digits) 0 <= S <= P.                                                                                                                                                                                                                                  | 16-byte `bytes` representing a little-endian 128-bit integer, to be divided by 10^S to get the decimal value
+| STRUCT&lt;T1,...,Tn&gt;       | A list of types in a defined order.                                                                                                                                                                                                                                                                                                                                   | `repeated Literal`, types matching T1..Tn
+| NSTRUCT&lt;N:T1,...,N:Tn&gt;  | **Pseudo-type**: A struct that maps unique names to value types. Each name is a UTF-8-encoded string. Each value can have a distinct type. Note that NSTRUCT is actually a pseudo-type, because Substrait's core type system is based entirely on ordinal positions, not named fields. Nonetheless, when working with systems outside Substrait, names are important. | n/a
+| LIST&lt;T&gt;                 | A list of values of type T. The list can be between [0..2,147,483,647] values in length.                                                                                                                                                                                                                                                                              | `repeated Literal`, all types matching T
+| MAP&lt;K, V&gt;               | An unordered list of type K keys with type V values. Keys may be repeated. While the key type could be nullable, keys may not be null.                                                                                                                                                                                                                                | `repeated KeyValue` (in turn two `Literal`s), all key types matching K and all value types matching V
+| PRECISIONTIMESTAMP&lt;P&gt;   | A timestamp with fractional second precision (P, number of digits) <= 9. Does not include timezone information and can thus not be unambiguously mapped to a moment on the timeline without context. Similar to naive datetime in Python.                                                                                                                             | `uint64` nanoseconds since 1970-01-01 00:00:00.000000000 (in an unspecified timezone)
```

Comment:
This looks like `PRECISIONTIMESTAMP` has a literal representation of `uint64 nanoseconds` and `PRECISIONTIMESTAMPTZ` has a literal representation of `uint64 microseconds`.

However, if I'm reading the proto correctly, they both have a literal representation of "either uint64 microseconds or nanoseconds since ..."